ingest-v2/src/main/kotlin/com/microsoft/azure/kusto/ingest/v2/container/BlobUploadContainer.kt

```diff
@@ -20,114 +19,378 @@ import com.microsoft.azure.kusto.ingest.v2.UPLOAD_RETRY_TIMEOUT_SECONDS
 import com.microsoft.azure.kusto.ingest.v2.common.ConfigurationCache
 import com.microsoft.azure.kusto.ingest.v2.common.exceptions.IngestException
 import com.microsoft.azure.kusto.ingest.v2.models.ContainerInfo
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.coroutineScope
 import org.slf4j.LoggerFactory
 import java.io.InputStream
 import java.time.Duration
+import java.time.OffsetDateTime
+import java.util.concurrent.atomic.AtomicInteger
 
 private val DEFAULT_RETRY_OPTIONS =
```

Comment:
This variable is unused right now, needs to be used in upload ?

Reply:
Since, we already have default retries in BlobClientBuilder in Azure Blob sdk level(4 attempts with exponential backoff), so I implemented a simple retry(max Retries = 3) with container rotation and switches to a different container on failure. So no need of these complex retry option anymore, removing it.

Reply:
Sure.. One quick question Tanmaya.. do we know if we have pass some options to BlobContainer or a similar construct to do it or is it something that is documented.

Any reference would be great.

Reply:
not sure if it defined in the official docs, I had checked in the Blob sdk code, BlobUploadContainer.kt------->(AZURE BLOB SDK starts)BlobClientBuilder->buildClient()->constructPipeline()->BuilderHelper.buildPipeline()->BuilderUtils.createRetryPolicy()->return new RequestRetryPolicy(retryOptions)->RequestRetryOptions.java

In RequestRetryOptions.java

this.retryPolicyType = retryPolicyType == null ? **RetryPolicyType.EXPONENTIAL** : retryPolicyType;
        if (maxTries != null) {
            StorageImplUtils.assertInBounds("maxRetries", (long)maxTries, 1L, 2147483647L);
            this.maxTries = maxTries;
        } else {
            **this.maxTries = 4;**
        }